runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -62,19 +63,32 @@
     /**
      * A cache for read results.
      */
-    final Cache<Long, ILogData> readCache = CacheBuilder.newBuilder()
-            .maximumSize(runtime.getParameters().getNumCacheEntries())
-            .expireAfterAccess(runtime.getParameters().getCacheExpiryTime(), TimeUnit.SECONDS)
-            .expireAfterWrite(runtime.getParameters().getCacheExpiryTime(), TimeUnit.SECONDS)
-            .removalListener(this::handleEviction)
-            .recordStats()
-            .build();
+    final Cache<Long, ILogData> readCache;
 
     /**
      * Constructor for the Address Space View.
      */
     public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
         super(runtime);
+
+        CacheBuilder cacheBuilder = CacheBuilder.newBuilder();
+
+        if (runtime.getParameters().getMaxCacheEntries() != 0) {
+            cacheBuilder.maximumSize(runtime.getParameters().getMaxCacheEntries());
+        }
+
+        if (runtime.getParameters().getMaxCacheWeight() != 0) {
+            cacheBuilder.maximumWeight(runtime.getParameters().getMaxCacheWeight());
+            cacheBuilder.weigher((k, v) -> (int) (MetricsUtils.sizeOf.deepSizeOf(k)
```

Comment:
Probably can hardcode the size of the key, it would be cheaper.

Reply:
Can we do this?

Reply:
Done.